lib/floe/workflow/runner/kubernetes.rb

```diff
@@ -51,72 +111,94 @@ def pod_name(image)
           "#{container_short_name}-#{SecureRandom.uuid}"
         end
 
-        def pod_spec(image, env, secret = nil)
-          container_spec = {
-            "name"  => container_name(image),
-            "image" => image,
-            "env"   => env.to_h.map { |k, v| {"name" => k, "value" => v.to_s} }
+        def pod_spec(name, image, env, secret = nil)
+          spec = {
+            :kind       => "Pod",
+            :apiVersion => "v1",
+            :metadata   => {
+              :name      => name,
+              :namespace => namespace
+            },
+            :spec       => {
+              :containers    => [
+                {
+                  :name  => container_name(image),
+                  :image => image,
+                  :env   => env.to_h.map { |k, v| {:name => k, :value => v.to_s} }
```

Comment:
Is the `.to_h` necessary here?  The only thing I think it will ensure is that there aren't duplicate keys, but a .map should work the same on a Hash or an associative Array.